docs/pages/getting-started/5-irma-contract.rst

```diff
@@ -0,0 +1,343 @@
+        sessionToken:    r => r.sessionID
+      }
+    }
+
+The ``session`` object contains all the technical parts to connect the IRMA JS library to your backend.
+The contents of the ``start`` object configures the initial request to start a signing session. You can control the type of request and the contents.
+In this case, some data from the frontend is sent as JSON. This is optional and no particular data is required.
+The ``url``, in this case ``/web/auth``, must be set in such a way that two backend services are available to the frontend.
+
+.. code-block::
+
+    <url>/session
+    <url>/session/<sessionToken>/result
+
+must both be available on the backend. For the example above this means that both ``/web/auth/session/`` and ``/web/auth/session/<sessionToken>/result`` are available. The ``<sessionToken>`` is the token that will be returned by the call to ``<url>/session/``.
+How to parse the result of that call and extract the token is done via the ``mapping`` object.
+
+The ``mapping`` object is a map where two keys are expected: ``sessionPtr`` and ``sessionToken``.
+``sessionPtr`` must point to the data that is used to render the QR code.
+``sessionToken`` must point to the session token used to get the result.
+
+Setting up the backend
+**********************
+
+As discussed in the previous chapter, the backend is required to expose two APIs to the frontend:
+
+.. code-block::
+
+    <url>/session
+    <url>/session/<sessionToken>/result
+
+No particular security context is required, you may require a user session if needed.
+
+Starting a session
+==================
+
+The ``<url>/session`` API is used to start a session.
+To start a session at the Nuts node, a valid contract has to be drawn up first.
+You can create such a contract with the following API on the Nuts node:
+
+.. code-block::
+
+    PUT /internal/auth/v1/contract/drawup
+
+With the following body:
+
+.. code-block:: json
+
+    {
+      "type": "BehandelaarLogin",
+      "language": "NL",
+      "version": "v3",
+      "legalEntity": "did:nuts:90348275fjasihnva4857qp39hn",
+      "validFrom": "2006-01-02T15:04:05+02:00",
+      "validDuration": "2h"
+    }
+
+The ``type`` must be one of the valid Nuts contract types, currently only ``BehandelaarLogin`` for Dutch and ``PractitionerLogin`` for English are supported.
+The `language`` selects the correct language, ``NL`` for Dutch and ``EN`` for english. The ``version`` must be v3.
+The legalEntity must refer to the DID of the current organization. The user either selects an organization to login for or is already logged in.
+From that context the organization must have a DID as described in :ref:`Getting Started on customer integration <connecting-crm>`.
+``validFrom`` is a RFC3339 compliant time string. ``validDuration`` describes how long the contract is valid for. Unix compatible duration strings can be used like ``1h`` or ``60m``. The local system timezone is used to format the date and time string.
```

Comment:
I don't think "Unix duration strings" are a thing? In any case add a link.

I'd guess we're using Golang's `Duration` format; https://golang.org/pkg/time/#ParseDuration

Reply:
I added all valid time units.